### DISCUSSION TITLE
Adjust mqueue/msg_max for container under MacOS

### DISCUSSION THREAD

**Discussion by DavidEdell** [containers/podman#20043] [Q&A] [answered]
I'm trying to run software in a container that makes heavy usage of POSIX message queues.  It fails to run from within the container, even if using `--privileged`.  Using arguments such as `--sysctl 'fs.mqueue.msg_max=256'` result in an "OCI permission denied" error.

Interestingly, the application does run without issue from within the podman machine.  ie: `podman machine ssh` and then running the application from the mounted folder.  From within the podman machine I also get permission denied errors if I try altering `/proc/sys/fs/mqueue/msg_max` from there.

Any ideas?  Thanks.

**Comment by Luap99** [comment, marked as answer]
see https://github.com/containers/podman/discussions/19737

rootless users cannot adjust the limt, if you create the machine with `podman machine init --rootful` it should work.

**Reply by DavidEdell** [reply]
That did the trick, thanks.

For future reference:

- podman machine stop && podman machine rm
- podman machine init --rootful
- ... rebuild container
- podman run -it --rm --sysctl 'fs.mqueue.msg_max=256' -v ${PWD}:/shared myImage
